numba/tests/test_typedlist.py

```diff
@@ -479,6 +480,144 @@ def foo():
         )
 
 
+class TestNoneType(MemoryLeakMixin, TestCase):
+
+    def test_append_none(self):
+        @njit
+        def impl():
+            l = List()
+            l.append(None)
+            return l
+
+        self.assertEqual(impl.py_func(), impl())
+
+    def test_len_none(self):
+        @njit
+        def impl():
+            l = List()
+            l.append(None)
+            return len(l)
+
+        self.assertEqual(impl.py_func(), impl())
+
+    def test_getitem_none(self):
+        @njit
+        def impl():
+            l = List()
+            l.append(None)
+            return l[0]
+
+        self.assertEqual(impl.py_func(), impl())
+
+    def test_setitem_none(self):
+        @njit
+        def impl():
+            l = List()
+            l.append(None)
+            l[0] = None
+            return l
+
+        self.assertEqual(impl.py_func(), impl())
+
+    def test_equals_none(self):
+        @njit
+        def impl():
+            l = List()
+            l.append(None)
+            m = List()
+            m.append(None)
+            return l == m, l != m, l < m, l <= m, l > m, l >= m
+
+        self.assertEqual(impl.py_func(), impl())
+
+    def test_not_equals_none(self):
+        @njit
+        def impl():
+            l = List()
+            l.append(None)
+            m = List()
+            m.append(1)
+            return l == m, l != m, l < m, l <= m, l > m, l >= m
+
+        self.assertEqual(impl.py_func(), impl())
+
+    def test_iter_none(self):
+        @njit
+        def impl():
+            l = List()
+            l.append(None)
+            l.append(None)
+            l.append(None)
+            count = 0
+            for i in l:
+                count += 1
+            return count
+
+        self.assertEqual(impl.py_func(), impl())
+
+    @unittest.skip("Works only with _disable_reflected_list")
+    def test_square_bracket_builtin_with_None(self):
+        @njit(_disable_reflected_list=True)
+        def foo():
+            l = [None, None, None]
+            return l
+        expected = List()
+        expected.append(None)
+        expected.append(None)
+        expected.append(None)
+        received = foo()
+        self.assertEqual(expected, received)
+
+    @unittest.skip("Works only with _disable_reflected_list")
+    def test_list_comprehension_with_none(self):
+        @njit(_disable_reflected_list=True)
+        def foo():
+            l = List()
+            # the following construct results in a List[None] that is discarded
+            [l.append(i) for i in (1, 3, 3)]
+            return l
+        expected = List()
+        [expected.append(i) for i in (1, 3, 3)]
+        received = foo()
+        self.assertEqual(expected, received)
```

Comment:
Given this flag doesn't exist, and may never exist in this form, should these be present?

Reply:
I can delete them, but if the flag does end up coming back, the tests will be needed, so I'll save them in branch for now.

Reply:
Sounds good, there's a comment here https://github.com/numba/numba/pull/5151/files#diff-3c7633fb37583b4d8a2f940fa1c73339R595-R596 that's of a similar ilk.

Reply:
also removed